tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala

```diff
@@ -89,10 +89,17 @@ object VarDictJava extends Configuration {
   /** Plain text tumor-normal separator. */
   val TumorNormalSeparator: String = "|"
 
-  /** Pulls a sample name out of a BAM file. */
-  private[vc] def extractSampleName(bam: PathToBam): String = {
+  /** Return the first read group in a BAM. */
+  private[vc] def firstReadGroup(bam: PathToBam): Option[SAMReadGroupRecord] = {
+    import com.fulcrumgenomics.commons.CommonsDef.javaIteratorAsScalaIterator
     val in = SamReaderFactory.make().open(bam)
-    yieldAndThen(in.getFileHeader.getReadGroups.iterator().next().getSample) { in.close() }
+    yieldAndThen(in.getFileHeader.getReadGroups.iterator.toStream.headOption) { in.close() }
+  }
+
+  /** Return the sample name from the first read group in a BAM. */
```

Comment:
Perhaps emphasize the first read group _**must exist**_, but the sample name does not need to exist.